config/software/ruby.rb

```diff
@@ -18,238 +18,31 @@
 
 # This is now the main software project for anything ruby related.
 # Even if you want a pre-built version of ruby from ruby-installer, include
-# this project as a dependency. If the version is set to ruby-windows,
-# it redirects to depend on the ruby-windows project.
+# this software as a dependency. It redirects to the appropriate "concrete"
+# ruby software definition based on the requested build/platform configuration.
 
 if windows?
-  default_version "ruby-windows"
+  default_version "rubyinstaller"
 else
-  # - chef-client cannot use 2.2.x yet due to a bug in IRB that affects chef-shell on linux:
-  #   https://bugs.ruby-lang.org/issues/11869
-  # - the current status of 2.3.x is that it downloads but fails to compile.
-  default_version "2.1.6"
+  default_version "compiled"
 end
 
-fips_enabled = (project.overrides[:fips] && project.overrides[:fips][:enabled]) || false
-
-if windows? && version == "ruby-windows"
-    dependency "ruby-windows"
-    dependency "ruby-windows-devkit"
-    dependency "ruby-windows-devkit-bash"
-    # The custom yakyakyak ruby build comes with openssl and the FIPS module.
-    # Don't clobber it.
-    dependency "openssl-windows" unless fips_enabled
-    dependency "cacerts"
-else
-
-if windows?
-  dependency "mingw"
-  dependency "patch"
+dependency "cacerts"
+
+case version
+when "rubyinstaller"
+  dependency "ruby-rubyinstaller"
+  dependency "openssl-windows"
+when "jaym-fips"
```

Comment:
I'm leaving the option to use that here just in case the sky is falling and we need to go back and repro some issue with SAP's first chef-client build.  No pipeline is currently using it.